esp-hal/Cargo.toml

```diff
@@ -1,19 +1,20 @@
 [package]
-name         = "esp-hal"
-version      = "0.15.0"
-edition      = "2021"
-rust-version = "1.76.0"
-description  = "Bare-metal HAL for Espressif devices"
-repository   = "https://github.com/esp-rs/esp-hal"
-license      = "MIT OR Apache-2.0"
-links        = "esp-hal"
+name          = "esp-hal"
+version       = "0.16.0"
+edition       = "2021"
+rust-version  = "1.76.0"
+description   = "Bare-metal HAL for Espressif devices"
+documentation = "https://docs.esp-rs.org/esp-hal/"
+repository    = "https://github.com/esp-rs/esp-hal"
+license       = "MIT OR Apache-2.0"
+links         = "esp-hal"
 
 [dependencies]
 bitflags             = "2.4.2"
 bitfield             = "0.14.0"
 cfg-if               = "1.0.0"
 critical-section     = "1.1.2"
-defmt                = { version = "=0.3.5", optional = true }
+defmt                = { version = "0.3.6", optional = true }
```

Comment:
I think pinning dependencies is annoying and can cause problems. I'd rather not, others can weigh in I guess. If they're breaking things on patch releases that makes me much less inclined to use their libraries.

Reply:
I'd mostly agree, but defmt does seem to break often, we already pin it in espflash, so I think we probably should pin it here?

Reply:
Alright, well that's incredibly annoying but I guess I'll re-pin it.

Reply:
I don't have any strong opinion here, I think we should be good not having it pinned. Just wanted to raise the question in case the change slipped in without us noticing it (this already happened to me in the past using the “Update all crates” command from the `crates` VS Code extension)

Reply:
Has anybody actually tested `defmt` with the lastest `main`? I mentioned that it was not being tested in CI at all but did not hear of anything after that.

Reply:
I _think_ I tested defmt-espflash for this PR: https://github.com/esp-rs/no_std-training/pull/37
Not the latest commit but recent enough I guess